src/cache/cache_las.c

```diff
@@ -546,19 +632,33 @@ __wt_las_insert_block(WT_SESSION_IMPL *session, WT_CURSOR *cursor,
 		} while ((upd = upd->next) != NULL);
 	}
 
+err:	/* Resolve the transaction. */
+	if (local_txn) {
+		if (ret == 0)
+			ret = __wt_txn_commit(las_session, NULL);
+		else
+			WT_TRET(__wt_txn_rollback(las_session, NULL));
+	}
+
+	__las_pop_isolation(las_session, saved_isolation);
+
 	if (insert_cnt > 0) {
-		(void)__wt_atomic_add64(
-		    &S2C(session)->cache->las_entry_count, insert_cnt);
-		WT_ERR(__las_insert_block_verbose(session, multi));
+		if (ret == 0) {
```

Comment:
While this makes sense with careful reading, a comment would be good here.

Reply:
I went with:
```
If the transaction successfully committed and we inserted records,
adjust the final entry count. We may have also deleted records, 
but we must have intended to insert records to be in this function
at all, checking the insert count is sufficient.
```